File: patient-service/src/main/java/org/example/patientservice/entity/Patient.java
package org.example.patientservice.entity;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import java.sql.Timestamp;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "patients")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Patient {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "patient_id")
    private Integer patientId;

    @Column(name = "user_id", unique = true)
    private Integer userId;

    @Column(name = "identity_number", unique = true, length = 20)
    private String identityNumber;

    @Column(name = "insurance_number", unique = true, length = 20)
    private String insuranceNumber;

    @Column(name = "first_name", length = 100)
    private String firstName;

    @Column(name = "last_name", length = 100)
    private String lastName;

    private LocalDate birthday;

    @Enumerated(EnumType.STRING)
    private Gender gender;

    @Column(columnDefinition = "TEXT")
    private String address;

    @Column(columnDefinition = "TEXT")
    private String allergies;

    private Integer height;

    private Integer weight;

    @Column(name = "blood_type", length = 10)
    private String bloodType;

    @Column(name = "created_at")
    @CreationTimestamp
    private Timestamp createdAt;

    public enum Gender {
        MALE, FEMALE, OTHER
    }

    @OneToMany(mappedBy = "patient", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference
    private List<EmergencyContact> emergencyContacts = new ArrayList<>();
}

File: frontend/web/src/role-doctor/types/patient.ts
import { Appointment } from "./appointment"
import { Schedule } from "./schedule"
import { AppointmentNote } from "./appointmentNote"
import { DoctorInfo } from "./doctor"

export interface Patient {
  id: number
  name: string
  code: string
  appointment: string
  date: string
  gender: string
  age: number
  symptom: string
  status: string
  avatar?: string
  appointmentData?: Appointment
}

export interface PatientInfo {
  avatar: string,
  patientId: number;
  userId: number;
  identityNumber: string;
  insuranceNumber: string;
  fullName: string;
  birthday: string; 
  gender: 'MALE' | 'FEMALE'
  address: string;
  allergies?: string | "Không xác định";
  height?: number | "Không xác định" ;
  weight?: number | "Không xác định";
  bloodType?: string | "Không xác định";
  createdAt: string; 
}

export interface PatientDetail {
  appointmentId: string
  doctorId: number
  schedule: Schedule
  symptoms: string,
  number: number,
  appointmentStatus: string,
  createdAt: string,
  patientInfo: PatientInfo
  doctorInfo: DoctorInfo
  appointmentNote: AppointmentNote
  slotEnd: string
  slotStart: string
}

export interface PatientDetail2 {
  appointmentId: string
  patientId: string
  patientInfo: PatientInfo
  symptoms: string,
  number: number,
  slotEnd: string
  slotStart: string
  doctorId: number
  schedule: Schedule
  appointmentStatus: string,
  
  
  createdAt: string,
  
  doctorInfo: DoctorInfo
  appointmentNote: AppointmentNote
  
}